### PR TITLE
fix(sdk): Remove event listeners on unsubscribe

### PR DESCRIPTION
**Pull request by jhermsmeier**
Change-Type: patch

### CODE REVIEW

File: lib/shared/sdk/index.js [hunk 1]
```diff
@@ -231,7 +230,10 @@ SDK.Scanner = class Scanner extends EventEmitter {
       return this
     }
 
-    this.adapters.delete(instance.name)
+    instance.removeListener('devices', this.onDevices)
+    instance.removeListener('error', this.onError)
+
+    this.adapters.delete(instance.id)
     this.emit('unsubsribe', adapter)
```
Comment (lurch) [inline lib/shared/sdk/index.js]: `unsubsribe`?? Does this PR need to be rebased?

Comment (jhermsmeier) [reply lib/shared/sdk/index.js]: The typo-correction PR isn't merged yet, will rebase whichever one comes after the other

Comment (lurch) [reply lib/shared/sdk/index.js]: Aha. I've just merged #1832 for you :)

Comment (jhermsmeier) [reply lib/shared/sdk/index.js]: Sweet, rebased that now

**Pull request merged**